fighters/lucas/src/acmd/aerials.rs

```diff
@@ -18,22 +18,23 @@ unsafe fn lucas_attack_air_n_game(fighter: &mut L2CAgentBase) {
-        ATTACK(fighter, 4, 0, Hash40::new("kneer"), 2.0, 365, 50, 0, 40, 3.5, 1.0, 0.0, 0.0, None, None, None, 0.5, 1.4, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_POS, false, 0, 0.0, 4, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_elec"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_ELEC, *ATTACK_REGION_PSI);
+        ATTACK(fighter, 0, 0, Hash40::new("top"), 1.4, 365, 48, 0, 34, 4.0, 0.0, 8.0, 3.5, None, None, None, 0.5, 1.2, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_POS, false, -1, 0.0, 3, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_elec"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_ELEC, *ATTACK_REGION_PSI);
+        ATTACK(fighter, 1, 0, Hash40::new("top"), 1.4, 365, 48, 0, 34, 4.0, 0.0, 8.0, -3.1, None, None, None, 0.5, 1.2, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_POS, false, -1, 0.0, 3, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_elec"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_ELEC, *ATTACK_REGION_PSI);
+        ATTACK(fighter, 2, 0, Hash40::new("top"), 1.4, 365, 48, 0, 34, 4.0, 0.0, 2.0, 3.5, None, None, None, 0.5, 1.2, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_POS, false, -1, 0.0, 3, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_elec"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_ELEC, *ATTACK_REGION_PSI);
+        ATTACK(fighter, 3, 0, Hash40::new("top"), 1.4, 365, 48, 0, 34, 4.0, 0.0, 2.0, -3.1, None, None, None, 0.5, 1.2, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_POS, false, -1, 0.0, 3, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_elec"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_ELEC, *ATTACK_REGION_PSI);
+        //ATTACK(fighter, 4, 0, Hash40::new("kneer"), 2.0, 365, 50, 0, 40, 3.5, 1.0, 0.0, 0.0, None, None, None, 0.5, 1.4, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_POS, false, 0, 0.0, 4, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_elec"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_ELEC, *ATTACK_REGION_PSI);
+        //ATK_SET_SHIELD_SETOFF_MUL_arg5(fighter, 0, 1, 2, 3, 0.0);
     }
-    frame(lua_state, 25.0);
+    frame(lua_state, 23.0);
     if is_excute(fighter) {
-        MotionModule::set_rate(boma, 1.0);
         AttackModule::clear_all(boma);
+        MotionModule::set_rate(boma, 3.0);
     }
     frame(lua_state, 26.0);
     if is_excute(fighter) {
         MotionModule::set_rate(boma, 1.0);
```

Comment:
lets convert these MotionModule::set_rate calls to the standard `FT_MOTION_RATE(fighter, C / (B - A))` format

`MotionModule::set_rate(boma, 3.0);` --> `FT_MOTION_RATE(fighter, 1.0 / (26.0 - 23.0));` for example

Reply:
also, you shouldn't wrap them in `is_excute`

Reply:
i actually think using FT_MOTION_RATE for lucas will be unfeasable because he uses an anim double jump, and rn what FT's doing is speeding the whole fighter up as opposed to just the animation, so rn if i cancel the move while in the sped up parts, i just shoot up, which MotionModule call completely doesnt do so

putting the MotionModule calls outside of the if_execute worked though, i can push that now, but if you still REALLY want those numbers in your formula ima need some help cuz i really dont know what to do with some of these raw doubles...

Reply:
alright, i made a new push addressed this, updating body to reflect changes soon

BIG thanks to jaccob101 🙏🙏